devel/onetbb/files/patch-onetbb-older-platforms.diff

```diff
@@ -0,0 +1,513 @@
+ # you may not use this file except in compliance with the License.
+@@ -29,6 +29,7 @@ ___TBB_malloc_safer_free
+ ___TBB_malloc_safer_msize
+ ___TBB_malloc_safer_realloc
+ ___TBB_malloc_free_definite_size
++
+ /* memory pool stuff */
+ __ZN3rml11pool_createElPKNS_13MemPoolPolicyE
+ __ZN3rml14pool_create_v1ElPKNS_13MemPoolPolicyEPPNS_10MemoryPoolE
+diff --git a/src/tbbmalloc_proxy/proxy_overload_osx.h b/src/tbbmalloc_proxy/proxy_overload_osx.h
+index 69582983..942933e2 100644
+--- a/src/tbbmalloc_proxy/proxy_overload_osx.h
++++ b/src/tbbmalloc_proxy/proxy_overload_osx.h
+@@ -1,5 +1,5 @@
+ /*
+-    Copyright (c) 2005-2022 Intel Corporation
++    Copyright (c) 2005-2023 Intel Corporation
+ 
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+@@ -134,10 +134,14 @@ struct DoMallocReplacement {
+         introspect.force_lock = &zone_force_lock;
+         introspect.force_unlock = &zone_force_unlock;
+         introspect.statistics = zone_statistics;
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__POWERPC__) // zone_locked is not defined on macOS < 10.6 and any PPC
+         introspect.zone_locked = &zone_locked;
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1070 // Only available on macOS > 10.6
+         introspect.enable_discharge_checking = &impl_zone_enable_discharge_checking;
+         introspect.disable_discharge_checking = &impl_zone_disable_discharge_checking;
+         introspect.discharge = &impl_zone_discharge;
++    #endif
++    #endif
+ 
+         zone.size = &impl_malloc_usable_size;
+         zone.malloc = &impl_malloc;
+@@ -150,8 +154,10 @@ struct DoMallocReplacement {
+         zone.introspect = &introspect;
+         zone.version = 8;
+         zone.memalign = impl_memalign;
++    #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__POWERPC__) // Unavailable on macOS < 10.6 and any PPC
+         zone.free_definite_size = &impl_free_definite_size;
+         zone.pressure_relief = &impl_pressure_relief;
++    #endif
+ 
+         // make sure that default purgeable zone is initialized
+         malloc_default_purgeable_zone();
+diff --git a/test/common/doctest.h b/test/common/doctest.h
+index 3b906764..b7116596 100644
+--- a/test/common/doctest.h
++++ b/test/common/doctest.h
+@@ -1,5 +1,5 @@
+ /*
+-    Modifications Copyright (c) 2020-2022 Intel Corporation
++    Modifications Copyright (c) 2020-2023 Intel Corporation
+     Modifications Licensed under the Apache License, Version 2.0;
+     You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+@@ -412,6 +412,9 @@ DOCTEST_MSVC_SUPPRESS_WARNING(26812) // Prefer 'enum class' over 'enum'
+ #elif defined(DOCTEST_PLATFORM_MAC)
+ #if defined(__x86_64) || defined(__x86_64__) || defined(__amd64__) || defined(__i386)
+ #define DOCTEST_BREAK_INTO_DEBUGGER() __asm__("int $3\n" : :) // NOLINT (hicpp-no-assembler)
++#elif defined(__POWERPC__)
```

Comment:
Use `__ppc__`.

Reply:
Here we need `__POWERPC__` (or otherwise `__ppc__ && __ppc64__`, but that’s longer code).

Reply:
It seems `__POWERPC__` in caps is [not available on Linux](https://github.com/gcc-mirror/gcc/blob/fe4608efc15b881ac908a3f90d7322736495ae72/gcc/config/rs6000/linux64.h#L292) and other platforms when using GCC. Since you plan to upstream this, it seems it's best to use `__PPC__` when you want to handle both 32-bit and 64-bit.

Reply:
> Then we need both, since `__PPC__` is not defined on Apple.

That's what I was worried about. Then I guess either add a check for `__APPLE__` as well to be explicit like in other places, or if this change applies to PPC in general, check for both `__PPC__` and `__POWERPC__`.

Reply:
`__POWERPC__` is already Apple-specific (well, it is also defined on BeOS, but nowhere else). I am pretty sure not a single person ever used Macports on BeOS :) 
But we can add a check for Apple too, that won’t hurt.

Reply:
You are right though, let’s make it transparent, otherwise this is nowhere obvious.

Reply:
@mohd-akram In fact it is fine as-is, as I now noticed (missed that when browsing from mobile): condition is wrapped in `#elif defined(DOCTEST_PLATFORM_MAC)`, so it is already Apple-only, and we do not need to repeat the check.